src/net/sourceforge/kolmafia/KoLmafia.java

```diff
@@ -1861,15 +1874,79 @@ public static String whoisPlayer(final String player) {
   public static boolean isPlayerOnline(final String player) {
     // This player is currently online.
     // This player is currently online in channel clan.
-    // This player is currently away from KoL in channel trade and listening to clan.
+    // This player is currently away from KoL in channel trade and listening to
+    // clan.
     String text = KoLmafia.whoisPlayer(player);
     return text != null && text.contains("This player is currently");
   }
 
   private static class UpdateCheckRunnable implements Runnable {
     public void run() {
-      // TODO: Check for new version on jenkins\github after migration is complete. See revision
-      // history for old release update check.
+      boolean checkUpdate = Preferences.getBoolean("checkRssUpdate");
+      if (!checkUpdate) {
+        return;
+      }
+
+      long lastUpdate = Long.parseLong(Preferences.getString("lastRssUpdate"));
+      long now = System.currentTimeMillis();
+      if (now - lastUpdate < 86400000L) {
+        return;
+      }
+
+      String contents = null;
+
+      try {
+        BufferedReader reader =
+            FileUtilities.getReader(
+                "https://api.github.com/repos/kolmafia/kolmafia/releases/latest");
+
+        contents = reader.lines().collect(Collectors.joining(System.lineSeparator()));
+        reader.close();
+      } catch (IOException e) {
+        System.out.println(e.getMessage() + " while trying to read from or close GitHub releases.");
+        return;
+      }
+
+      if (contents == null) {
+        return;
+      }
+
+      String currentVersion = null;
+
+      try {
+        JSONObject json = new JSONObject(contents);
+        currentVersion = (String) json.get("name");
+      } catch (JSONException e) {
+        System.out.println(e.getMessage() + " while loading JSON");
+        return;
+      }
+
+      if (currentVersion == null) {
+        return;
+      }
+
+      int currentVersionInt;
+      try {
+        currentVersionInt = Integer.parseInt(currentVersion) + 1;
+      } catch (NumberFormatException e) {
+        System.out.println(e.getMessage() + " while parsing version number");
+        return;
+      }
+
+      String lastVersion = Preferences.getString("lastRssVersion");
+      Preferences.setString("lastRssVersion", currentVersion);
+      Preferences.setString("lastRssUpdate", Long.toString(now));
+
+      if (currentVersionInt <= StaticEntity.getRevision() || currentVersion.equals(lastVersion)) {
+        return;
+      }
+
+      if (InputFieldUtilities.confirm(
+          String.format(
+              "A new version KoLmafia r%s is now available. Would you like to download it now?",
```

Comment:
formatting nit: perhaps "new version of KoLmafia (r%s) is now available"?  This reads a bit awkwardly to me as it is right now.

(Maybe that's just me)